### ISSUE TITLE
Compilation of optional FEM module fails with VS2019

### ISSUE THREAD

**Issue by fdgStilla** [InsightSoftwareConsortium/ITK#4399]
### Description
1. Compile with VS2019 the FEM module, we have some static assertions.

2. In addition if we add the -std:c++20 to the command line, we have the following error:
`Modules\Core\Common\include\itkImageRandomConstIteratorWithIndex.hxx(39): error C2244: 'itk::ImageRandomConstIteratorWithIndex<TImage>::SetNumberOfSamples': unable to match function definition to an existing declaration`

### Steps to Reproduce
1. activate `Module_ITKFEM` and `Module_ITKFEMRegistration` cmake variables
2. Compile with VS2019

**Comment by N-Dekker** [comment]
Thanks very much @fdgStilla, I did reproduce the VS2019 `-std:c++20` error C2244 messages, even while having the latest updates. As you wrote:

> error C2244: 'itk::ImageRandomConstIteratorWithIndex<TImage>::SetNumberOfSamples': unable to match function definition to an existing declaration

BTW, I got them even without trying to compile the FEM module!

**Comment by fdgStilla** [comment]
By the way if you want the whole story I am trying to use ITK with std:c++20 from conan, but my first attempt to patch the older versions was rejected and I was asked to use the last version of ITK instead: https://github.com/conan-io/conan-center-index/pull/22235. Note that the FEM is activated by default in conan recipe.

I also have another issue with the nifti third party, see here: https://github.com/NIFTI-Imaging/nifti_clib/issues/184, so I guess another PR in ITK will be needed.

**Comment by N-Dekker** [comment]
@fdgStilla I locally tried VS2019 `-std:c++20`, and encountered many more incorrect compile errors. Did you also encounter this one?

    error C2065: '__$ReturnUdt': undeclared identifier

🙀 

Shouldn't we just conclude that C++20 support of VS2019 is just too buggy to be used seriously? How about VS2022?

**Comment by fdgStilla** [comment]
I managed to compile with BUILD_TESTING=OFF (which is what conan will be doing I guess).

I managed to fix the "__$ReturnUdt" in LightProcessObject error by moving `SegmentationLevelSetFunction::AdvectionField`, it is some black magic. I did not go further, so I would recommend to create a dedicated issue to fully support c++20 with BUILD_TESTING.

We have been using VS2019 with c++20 for several years with all our projects without any issue so I would say it is not "too buggy".
I don't know about VS2022 support.
